core/modules/replica/WorkerServer.h

```diff
@@ -0,0 +1,152 @@
+  */
+class WorkerServer
+    : public std::enable_shared_from_this<WorkerServer>  {
+
+public:
+
+    /// The pointer type for instances of the class
+    typedef std::shared_ptr<WorkerServer> Ptr;
+
+    /**
+     * Static factory method is needed to prevent issue with the lifespan
+     * and memory management of instances created otherwise (as values or via
+     * low-level pointers).
+     *
+     * @param serviceProvider - for configuration, etc. services
+     * @param requestFactory  - the factory of requests
+     * @workerName            - the name of a worker this instance represents
+     *
+     * @return pointer to the new object created by the factory
+     */
+    static Ptr create(ServiceProvider::Ptr const& serviceProvider,
+                      WorkerRequestFactory& requestFactory,
+                      std::string const& workerName);
+
+    // Default construction and copy semantics are prohibited
+
+    WorkerServer() = delete;
+    WorkerServer(WorkerServer const&) = delete;
+    WorkerServer& operator=(WorkerServer const&) = delete;
+
+    ~WorkerServer() = default;
+
+    /// @return the name of a worker this server runs for
+    std::string const& worker() const { return _workerName; }
+
+    /**
+     * The processor object can be used for detailed monitoring of
+     * the on-going activities and statistics collection if needed.
+     *
+     * @return reference to the processor
+     */
+    WorkerProcessor const& processor() const { return _processor; }
+
+    /**
+     * Begin listening for and processing incoming connections
+     */
+    void run();
+
+private:
+
+    /**
+     * Construct the server with the specified configuration.
+     *
+     * @param serviceProvider - for configuration, etc. services
+     * @param requestFactory  - the factory of requests
+     * @workerName            - the name of a worker this instance represents
+     */
+    WorkerServer(ServiceProvider::Ptr const& serviceProvider,
+                 WorkerRequestFactory& requestFactory,
+                 std::string const& workerName);
+
+    /**
+     * Begin (asynchrnonously) accepting connection requests.
+     */
+    void beginAccept();
+    
+    /**
+     * Handle a connection request once it's detected. The rest of
+     * the comunication will be forewarded to the connection object
+     * specified as a parameter of the method.
+     *
+     * @param connection - object responsible for communications with a client
+     * @param ec         - error condition to be checked for
+     */
+    void handleAccept(WorkerServerConnection::Ptr const& connection,
+                      boost::system::error_code const& ec);
+
+    /// @return the context string
+    std::string context() const { return "SERVER  "; }
+
+private:
+
+    ServiceProvider::Ptr _serviceProvider;
+    std::string _workerName;
+
+    WorkerProcessor _processor;
+
+    /// Cached parameters of the worker
+    WorkerInfo const& _workerInfo;
```

Comment:
Well ... it's a bit complicated. Let's start with a statement that all these objects (of class WorkerInfo) have a longer lifespan than its clients because 1) they're provided by the Configuration service of the Replication framework, and 2) the Configuration service has been designed to preserve the objects through the lifetime of an application. That's all just technicality. An area where things are getting more complicated is in an intent to keep them by a reference rather than by a copy. Here an idea was to implement a dynamic reconfiguration of the application (say, a worker server) w/o needing to restart it. In that case I would need to extend each user of the Configuration service with an event-based mechanism to notify them (the users) about changes in the Configuration. My plan was to push these events to clients (including WorkerServer) so that they would re-evaluate their state and make proper adjustments to their behavior. That was the theory. In practice, the re-configuration (as well as the above mentioned notification mechanism) is yet to be implemented. Some of it is contingent to the K8 setup.
Having said that, I'm happy to eliminate the reference in this version of the code. The next iteration of the Replication system will take care of this.